orange/src/main/java/com/hartwig/hmftools/orange/algo/OrangeAlgo.java

```diff
@@ -658,25 +659,45 @@ private Map<PercentileType, Evaluation> evaluateCohortPercentiles(@NotNull Orang
     }
 
     @NotNull
-    private OrangePlots buildPlots(@NotNull OrangeConfig config) throws IOException
+    private OrangePlots buildPlots(@NotNull OrangeConfig config, LinxData linxData) throws IOException
     {
         LOGGER.info("Loading plots");
 
         plotManager.createPlotDirectory();
 
         String linxPlotDir = config.linxPlotDirectory();
         List<String> linxDriverPlots = Lists.newArrayList();
-        if(new File(linxPlotDir).exists())
+
+        if(linxPlotDir != null)
         {
-            for(String file : new File(linxPlotDir).list())
+            if(new File(linxPlotDir).exists())
             {
-                linxDriverPlots.add(plotManager.processPlotFile(linxPlotDir + File.separator + file));
+                for(String file : new File(linxPlotDir).list())
+                {
+                    linxDriverPlots.add(plotManager.processPlotFile(linxPlotDir + File.separator + file));
+                }
             }
+
+            Set<Integer> linxReportableClusters;
+            try
+            {
+                linxReportableClusters =
+                        LinxReportableClusters.findReportableClusters(linxData, config.linxSomaticDataDirectory(), config.tumorSampleId());
+            }
+            catch(IOException e)
+            {
+                throw new RuntimeException("Unable to determine linx reportable clusters", e);
+            }
+
```

Comment:
I would separate loading and evaluation. It's cleaner to add the new structures to LinxData. Even though that class exists in hmf-common, it is only used by ORANGE and already a bridge class holding all data from linx that is relevant for ORANGE.

Reply:
Ah, ok didn't realize LinxData was orange specific. That does make the particular data structures I chose (simple lists and maps of integer ids), seem untidy ... in the current implementation I can get it away with this since its private to a class with a simple public interface.

I'll start by moving the current structures to LinxData + LinxDataLoader as the common location does seem to make the data organization in orange nicer. Then can see if it looks like we'd want more conventional structures in LinxData.

Reply:
I'm fine with how LinxData currently looks. I haven't always mapped entire datamodels to their java representation in the past either, as long as it is clear what is meant by the datamodel.